### ISSUE TITLE
Add to package repositories

### ISSUE THREAD

**Issue by MitMaro** [MitMaro/git-interactive-rebase-tool#172]
I would nice to have this project available everywhere! I don't have the energy (or knowledge) to provide builds for everywhere the tool could be used. If anyone is willing to create a package I would be grateful!
 
**List taken from [ripgrep](https://github.com/BurntSushi/ripgrep#installation)**

* [x] Debian (.deb)
* [x] Homebrew
* [x] FreeBSD (pkg)
* [ ] Linuxbew
* [ ] MacPorts
* [x] Windows Chocolatey
* [x] Windows Scoop
* [x] Arch Linux (pacman)
* [ ] Gentoo (emerge)
* [ ] Fedora (dnf)
* [ ] openSUSE (zypper)
* [ ] Ubuntu (apt - PPA)
* [ ] OpenBSD (pkg_add)
* [ ] NetBSD (pkgin)

If anyone is willing to help please comment below!

**Comment by Rustin-Liu** [comment]
I will make a ppa for ubuntu and make a package for choco.

**Comment by Rustin-Liu** [comment]
The choco package waiting review :https://chocolatey.org/packages/git-interactive-rebase-tool/1.1.0

**Comment by MitMaro** [comment]
Thanks @Rustin-Liu !

**Comment by masaeedu** [comment]
I've created a PR to add this to Nixpkgs as well: https://github.com/NixOS/nixpkgs/pull/79078

**Comment by masaeedu** [comment]
We're observing [some failures](https://github.com/NixOS/nixpkgs/runs/421503909) related to the terminal color testcases. Unfortunately I don't really understand what the logic is doing since I'm not familiar with Rust, would you know what it means for the following tests to fail:

```
failures:
    display::utils::tests::detect_color_mode_no_env_2_colors
    display::utils::tests::detect_color_mode_vte_version_0_36_00
    display::utils::tests::detect_color_mode_vte_version_greater_0_36_00

test result: FAILED. 171 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out
```

**Comment by masaeedu** [comment]
The specific failures are:

```
failures:
---- display::utils::tests::detect_color_mode_no_env_2_colors stdout ----
thread 'display::utils::tests::detect_color_mode_no_env_2_colors' panicked at 'assertion failed: `(left == right)`
  left: `TrueColor`,
 right: `TwoTone`', src/display/utils.rs:84:3
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
---- display::utils::tests::detect_color_mode_vte_version_0_36_00 stdout ----
thread 'display::utils::tests::detect_color_mode_vte_version_0_36_00' panicked at 'assertion failed: `(left == right)`
  left: `TwoTone`,
 right: `TrueColor`', src/display/utils.rs:168:3
---- display::utils::tests::detect_color_mode_vte_version_greater_0_36_00 stdout ----
thread 'display::utils::tests::detect_color_mode_vte_version_greater_0_36_00' panicked at 'assertion failed: `(left == right)`
  left: `TwoTone`,
 right: `TrueColor`', src/display/utils.rs:175:3

```

**Comment by masaeedu** [comment]
I've provisionally patched out those tests and the PR was merged. Once Nix CI catches up you should be able to install this tool via:

```
nix-env -iA nixpkgs.gitAndTools.git-interactive-rebase-tool
```

**Comment by MitMaro** [comment]
Thanks @masaeedu ! 

Please excuse any poor usage of terminology with the NixOS ecosystem. I know a bit about the OS and it's package management, but I am a newbie at best!

I took a look into the PR to add the package NixOS. I think the issue is that by default NixOS tests Rust based packages using `cargo test -- `. Since those particular tests that are failing depend on modifying the process' environment variables and that by default Rust runs tests in parallel, it causes tests to interfere with each other.

This is further backed up by the seemingly randomness of the tests that were failing in the PR you linked about.

To fix this the `cargo test` command should take a argument of `--test-threads=1`.

I attempted to fix this in nixpkgs, but I only have a passing knowledge of the nix package system. I was able to reproduce the build but I could not get the build to fail when the patch to remove the tests were removed. I think it may limited by the single core that I use in my VM. I am also unsure how to pass arguments to `cargo test` within the nix package expression.

Since this isn't clear that this is a requirement to run tests, I am going to use a configuration file for Cargo to enforce this requirement without the need for projects to manually limit the number of test thread.

**Comment by masaeedu** [comment]
@MitMaro Ah, thanks for clarifying. The way you typically do this is by modifying the `checkPhase` of the derivation, i.e. you'd just add an attribute:

```
checkPhase = ''
  <whatever you're supposed to do to run tests>
'';
```

to the derivation's Nix expression. I tried specifying `cargo test --test-threads=1`, but I got the following error:

```
error: Found argument '--test-threads' which wasn't expected, or isn't valid in this context
        Did you mean --test?

USAGE:
    cargo test --test <NAME>...

For more information try --help
```

**Comment by masaeedu** [comment]
Btw, I couldn't get the tests to fail locally either, but I was assuming this was because my terminal has true color, and did not suspect it had anything to do with parallelism.

**Comment by MitMaro** [comment]
@masaeedu , I was about to write that you need to do the command `cargo test -- --test-threads=1` but it looks like you already figured it out in the linked PR.

Thanks again for putting this in the NixOS package system!

**Comment by 3v1n0** [comment]
Pleas include snapcraft.io to your list, so that it can run in any distro and you can manage easily the releases (will be automatic from git tags), but you need to ask the "classic" snap permissions to make it work everywhere :)

**Comment by glensc** [comment]
you can add repologoy summary like [git-filter-repo](https://github.com/newren/git-filter-repo) does:
- https://github.com/newren/git-filter-repo/blob/v2.26.0/INSTALL.md#installation-via-package-manager

**Comment by MitMaro** [comment]
Closing this, as I only plan to support the handful of platforms that I use at the moment. I will allow the community maintain and/or add them to other package repositories.

**Comment by xan2622** [comment]
It would be nice if *Git Interactive Rebase Tool* was installable thanks to **Winget**. 

Btw, only the version 1.1.0 is available on Chocolatey: https://community.chocolatey.org/packages/git-interactive-rebase-tool
